File: 1663-smallest-string-with-a-given-numeric-value/1663-smallest-string-with-a-given-numeric-value.cpp
class Solution {
public:
    string getSmallestString(int n, int k) {
        return solve(n, k);
    }
private:
    string solve(int n, int k) {
        string ans(n, 'a');
        for(int i = n - 1; i > -1; --i) {
            if((k - i) > 26) {
                ans[i] = 'z';
                k = k - 26;
            }
            else {
                ans[i] = 'a' + k - i - 1;
                k = i;
            }
        }
        return ans;
    }
};

File: 1663-smallest-string-with-a-given-numeric-value/1663-smallest-string-with-a-given-numeric-value.py
class Solution:
    def getSmallestString(self, n: int, k: int) -> str:
        ans = [0] * n
        
        for i in range(n - 1, -1, -1):
            if k-i > 26:
                ans[i] = 'z'
                k = k - 26
            else:
                ans[i] = chr(k - i + 96)
                k = i
            
        return ''.join(ans)